### PR TITLE
[now-next] Add `handle: miss` and `handle: hit` for custom-routes

### PR DESCRIPTION
**Pull request by ijjk**
This is required to match custom-routes behavior in Next.js by checking dynamic routes after each rewrite although is currently blocked on `now dev` also supporting the feature

This reverts commit 0da98a7f5d25f4c24e0d169bdcf30573859d25ed.

### CODE REVIEW

File: packages/now-next/src/utils.ts [hunk 1]
```diff
@@ -354,7 +354,7 @@ export async function getDynamicRoutes(
   dynamicPages: string[],
   isDev?: boolean,
   routesManifest?: RoutesManifest
-): Promise<{ src: string; dest: string }[]> {
+): Promise<{ src: string; dest: string; check: true }[]> {
```
Comment (styfle) [inline packages/now-next/src/utils.ts]:
```suggestion
): Promise<Source[]> {
```

You can do `import { Source } from '@now/routing-utils'` and use it everywhere you expect src/dest like this.

Comment (ijjk) [reply packages/now-next/src/utils.ts]: Updated

**Review by styfle — APPROVED** [review]
Thanks! 🎉

**Review by dav-is — APPROVED** [review]
LGTM

**Pull request merged**